### PR TITLE
Documentation changes for Laravel 6 upgrade

### PR DESCRIPTION
**Pull request by bennothommo**
Main PR: https://github.com/octobercms/october/pull/4893

Changes to the documentation of October CMS in preparation for the Laravel 6 upgrade.

### CODE REVIEW

File: services-events.md [hunk 1]
```diff
@@ -97,16 +97,19 @@ Sometimes you may wish to stop the propagation of an event to other listeners. Y
 <a name="wildcard-listeners"></a>
 ### Wildcard listeners
 
-When registering an event listener, you may use asterisks to specify wildcard listeners. The following listener will handle all events that begin with `foo.`.
+When registering an event listener, you may use asterisks to specify wildcard listeners. Wildcard listeners will receive the 
+event name fired first, followed by the parameters passed through to the event as an array.
```
Comment (LukeTowers) [inline services-events.md]:
```suggestion
When registering an event listener, you may use asterisks to specify wildcard listeners. Wildcard listeners will receive the event name fired first, followed by the parameters passed through to the event as an array.
```

File: database-collection.md [hunk 2]
```diff
@@ -32,46 +32,108 @@ However, collections are much more powerful than arrays and expose a variety of
             return $user->name;
         });
 
-<a name="usage-examples"></a>
-## Usage examples
```
Comment (LukeTowers) [inline database-collection.md]: Why remove this section?

File: database-relations.md [hunk 3]
```diff
@@ -410,6 +411,49 @@ Typical foreign key conventions will be used when performing the relationship's
...
+        'userHistory' => [
+            'Acme\Supplies\Model\History',
+            'key'        => 'supplier_id',
+            'through' => 'Acme\Supplies\Model\User'
+            'throughKey' => 'user_id',
+            'otherKey'   => 'id'
+        ],
+    ];
+
+
```
Comment (LukeTowers) [inline database-relations.md]: Wasn't there also another relationship type added?

File: help-unit-testing.md [hunk 4]
```diff
@@ -6,7 +6,7 @@
 <a name="testing-plugins"></a>
 ## Testing plugins
 
-Plugin unit tests can be performed by running `phpunit` in the base plugin directory.
+Plugin unit tests can be performed by running `vendor/bin/phpunit` in the base plugin directory.
```
Comment (LukeTowers) [inline help-unit-testing.md]:
```suggestion
Individual plugin test cases can be run by running `../../../vendor/bin/phpunit` in the plugin's base directory (ex. `plugins/acme/demo`.
```

File: database-collection.md (hunk 2 above)
Comment (bennothommo) [reply database-collection.md]: @LukeTowers It was renamed to "Available Methods" - more fitting for a bigger list of methods IMO.

File: database-relations.md (hunk 3 above)
Comment (bennothommo) [reply database-relations.md]: @LukeTowers nah, there was only HasOneThrough. I had added some missing code and documentation for the HasManyThrough as in the L6 branches, but that's already in the docs.

Comment (LukeTowers) [reply database-relations.md]: Sounds good @bennothommo

File: database-collection.md (hunk 2 above)
Comment (LukeTowers) [reply database-collection.md]: @bennothommo can you keep the anchor in place for existing links?

Comment (bennothommo) [reply database-collection.md]: @LukeTowers no worries - done.

File: services-helpers.md [hunk 5]
```diff
@@ -18,6 +18,7 @@ October includes a variety of "helper" PHP functions. Many of these functions ar
 <div class="collection-method-list" markdown="1">
 [array_add](#method-array-add)
 [array_divide](#method-array-divide)
+[array_undot](#method-array-undot)
```
Comment (LukeTowers) [inline services-helpers.md:21]: Shouldn't this come after array_dot in the menu?

Comment (bennothommo) [reply services-helpers.md]: Whoops - fixed.

**Pull request merged**